Matcher in loader, take 3

cro opened the pull request:
### What does this PR do?

Creates a new directory `salt/matchers` and migrates the matcher functions into separate .py files in this directory.  These functions are loaded at minion startup, but can be reloaded with the new `saltutil.refresh_matchers` function.

### Tests written?

Matcher tests apply here.

### Commits signed with GPG?

Yes

Commit:
modules/nilrt_ip.py: Add EtherCAT Support

The possibilities for adapter mode are: TCP/IP, EhterCAT and
Disabled.

Signed-off-by: Alexandru Vasiu <alexandru.vasiu@ni.com>
Signed-off-by: Ovidiu-Adrian Vancea <ovidiu.vancea@ni.com>

Commit:
modules/nilrt_ip.py: Add EtherCAT Support

The possibilities for adapter mode are: TCP/IP, EhterCAT and
Disabled.

Signed-off-by: Alexandru Vasiu <alexandru.vasiu@ni.com>
Signed-off-by: Ovidiu-Adrian Vancea <ovidiu.vancea@ni.com>

cbosdo committed:
Add virt.pool_info test case with no target element

The target element is not mandatory, introduce a test that we properly
handle this situation

cbosdo committed:
Remove minidom use in virt module

ElementTree provides a more convenient API than minidom. Switching to
ElementTree will reduce the number of dependencies for the virt module
and simplify the code a little.

Commit:
Deprecate 'master_shuffle' in favor of 'random_master'

cro committed:
Add matchers directory and move default set of matchers to separate .py

cro committed:
Add matchers directory and move default set of matchers to separate .py

cro committed:
Add matchers directory and move default set of matchers to separate .py

cro committed:
Add matchers directory and move default set of matchers to separate .py

cro committed:
Add matchers directory and move default set of matchers to separate .py

cro committed:
Add matchers directory and move default set of matchers to separate .py

cro committed:
Add some more needed imports

cro committed:
Add some more needed imports

cro committed:
Add some more needed imports

cro committed:
Add some more needed imports

cro committed:
Add some more needed imports

cro committed:
Handle this as a classmethod amd be smarter about setting it up.

cro committed:
Handle this as a classmethod amd be smarter about setting it up.

cro committed:
Handle this as a classmethod amd be smarter about setting it up.

cro committed:
Handle this as a classmethod amd be smarter about setting it up.

cro committed:
Handle this as a classmethod amd be smarter about setting it up.

cro committed:
Typo

cro committed:
Clarify setup

cro committed:
Clarify setup

cro committed:
Clarify setup

cro committed:
Clarify setup

cro committed:
Clarify setup

cro committed:
Lint

cro committed:
Lint

cro committed:
Lint

cro committed:
Lint

cro committed:
Lint

cro committed:
Remove unused imports

cro committed:
Remove unused imports

cro committed:
Split matchers out into separate .py files.  Add refresh_matchers, sync_matchers.

cro committed:
Split matchers out into separate .py files.  Add refresh_matchers, sync_matchers.

cro committed:
Split matchers out into separate .py files.  Add refresh_matchers, sync_matchers.

cro committed:
Split matchers out into separate .py files.  Add refresh_matchers, sync_matchers.

cro committed:
Split matchers out into separate .py files.  Add refresh_matchers, sync_matchers.

cro committed:
Refactor matchers to all use similar function signatures.

cro committed:
Refactor matchers to all use similar function signatures.

cro committed:
Refactor matchers to all use similar function signatures.

cro committed:
Refactor matchers to all use similar function signatures.

Commit:
modules/nilrt_ip.py: Fix configparser compatibility

read_file is available only since python 3.2 so readfp should
be used instead

Signed-off-by: Alexandru Vasiu <alexandru.vasiu@ni.com>

cbosdo committed:
Add virt.update function

User need to be able to update an existing virtual machine definition.
This function changes the definition for the next start of the VM and
tries hard to live update the virtual machine.

Commit:
add service enabled test

cro committed:
Add docs for loadable matchers.

Arabus committed:
Add query current minion config with config.items

Fixes #48680

cro committed:
Add master-side cache matcher

cro committed:
Add compound pillar_exact_match

cro committed:
Add details for master-side to the docs.

Commit:
modules/nilrt_ip.py: Clean-Up

Signed-off-by: Alexandru Vasiu <alexandru.vasiu@ni.com>

cro committed:
Lint

Ch3LL committed:
Fix test_managed_source_hash_indifferent_case on macosx to correct tmp path

cro committed:
Add matcher doc to Fluorine release notes, clean up some minor doc warnings that appear when building the docs.

cro committed:
restore orphan

cbosdo committed:
virt.init: allow 'none' graphics type

Graphics type 'none' will help to distinguish between no graphics data
and removal of graphics device in a future virt.update function.

cbosdo committed:
Updating running domains in virt.running

So far virt.running does nothing if the corresponding domain is already
defined. Use the new virt.update function to change the domain
configuration.

cbosdo committed:
Rename states/test_libvirt.py into states/test_virt.py

Modules test files are named after the module itself, the state virt
module needs to comply with that rule too.

mirceaulinic committed:
Add a few functions for network configuration manipulation

lomeroe committed:
pass proxy_user, proxy_password, proxy_host, and ca_certs through
salt.utils.stringutils.to_str()

tornado in py2 requires str not unicode strings

fixes #46896

garethgreenaway committed:
Fixing a case where the state module would fail if the state file being passed was all integers.  Added a new tests for this edge case.

Commit:
Add configuration test for minion_id_lowercase

Commit:
PEP8 E303: too many blank lines (3)

dwoz committed:
os.uname is not available on py2 windows

Commit:
Reduceded config, commented, dedented

dwoz committed:
Multiple windows test fixes

Commit:
Test lowercase on id_function

gtmanfred committed:
only do reverse dns lookup on ips for salt-ssh

Fixes #48676

Commit:
Added ability to create replica 3 arbiter 1 gluster volumes

cro committed:
Test fix

Commit:
Lint fix

Commit:
Updates requested by @rallytime

KaiSforza committed:
Only run lint checks against changed files

Commit:
Lint fix

Commit:
Fixing versionadded tags

KaiSforza committed:
test: except for OSError only

rallytime reviewed (approved):
This looks good to me!

I'd like a couple of other people to review it though.

KaiSforza committed:
Threshold was wrong

I also removed the delta value check. It's going to be weird with the
number of runs we do, and it's also set to run on failures.

cro commented:
I cannot make those tests fail for me "locally" (I've tried running them in a Docker container and on a machine in Amz).  
```
    integration.netapi.rest_tornado.test_app.TestSaltAPIHandler.test_multi_local_async_post
    integration.netapi.rest_tornado.test_app.TestSaltAPIHandler.test_multi_local_async_post_multitoken
    integration.netapi.rest_tornado.test_app.TestSaltAPIHandler.test_simple_local_async_post
    integration.netapi.rest_tornado.test_app.TestSaltAPIHandler.test_simple_local_post_only_dictionary_request_with_order_masters
    integration.netapi.rest_tornado.test_app.TestSaltAPIHandler.test_simple_local_runner_post
```

Are they known to be problematic tests?

Commit:
Fix salt.utils.process.py regression

Doing a pwd.getpwnam() will raise a KeyError
exception that is not catched anymore, after
a change that happened in 2012. Do catch that.

Signed-off-by: Rares POP <rares.pop@ni.com>

rallytime commented:
@cro I cleaned up the problematic tornado tests last week, so it's curious to me that they're failing consistently across all builds. That looks related to me.

Commit:
Reduce the number of days an issue is stale by 10

Commit:
modules/nilrt_ip.py: Fix argument for function _load_config

Signed-off-by: Alexandru Vasiu <alexandru.vasiu@ni.com>

cachedout commented:
Why were the tornado api tests modified here at all? Did something get pulled in unintetionally?

cro commented:
For this PR I went back and cherry-picked only my changes to make sure that nothing we didn't want was in here.  So I'm as confused as everyone, the above tests don't fail in a freshly-created environment.
